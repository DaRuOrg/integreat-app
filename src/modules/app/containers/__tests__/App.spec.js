import unnecessary

const mockStore = configureMockStore()
const mockedStore = mockStore()

jest.mock('../../createReduxStore', () => jest.fn().mockImplementation(() => mockedStore))

describe('App', () => {
  test('should render', () => {
    shallow(<App />)
  })

  test('should create correct store and pass it to Provider', () => {
    const app = shallow(<App />)

    expect(createReduxStore).toHaveBeenCalledWith(createHistory, {}, expect.any(Object))
    expect(app.find(Provider).prop('store')).toEqual(mockedStore)
  })
})
