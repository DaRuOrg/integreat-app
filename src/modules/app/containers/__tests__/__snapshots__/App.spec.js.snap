// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <EndpointProvider
    endpoints={
      Array [
        Endpoint {
          "_stateName": "languages",
          "finishFetchAction": [Function],
          "mapResponse": [Function],
          "mapStateToUrlParams": [Function],
          "responseOverride": undefined,
          "shouldRefetch": [Function],
          "startFetchAction": [Function],
          "url": "https://cms.integreat-app.de/{location}/{language}/wp-json/extensions/v0/languages/wpml",
        },
        Endpoint {
          "_stateName": "locations",
          "finishFetchAction": [Function],
          "mapResponse": [Function],
          "mapStateToUrlParams": [Function],
          "responseOverride": undefined,
          "shouldRefetch": [Function],
          "startFetchAction": [Function],
          "url": "https://cms.integreat-app.de/wp-json/extensions/v1/multisites",
        },
        Endpoint {
          "_stateName": "categories",
          "finishFetchAction": [Function],
          "mapResponse": [Function],
          "mapStateToUrlParams": [Function],
          "responseOverride": undefined,
          "shouldRefetch": [Function],
          "startFetchAction": [Function],
          "url": "https://cms.integreat-app.de/{location}/{language}/wp-json/extensions/v0/modified_content/pages?since=1970-01-01T00:00:00Z",
        },
        Endpoint {
          "_stateName": "disclaimer",
          "finishFetchAction": [Function],
          "mapResponse": [Function],
          "mapStateToUrlParams": [Function],
          "responseOverride": undefined,
          "shouldRefetch": [Function],
          "startFetchAction": [Function],
          "url": "https://cms.integreat-app.de/{location}/{language}/wp-json/extensions/v0/modified_content/disclaimer?since=1970-01-01T00:00:00Z",
        },
        Endpoint {
          "_stateName": "events",
          "finishFetchAction": [Function],
          "mapResponse": [Function],
          "mapStateToUrlParams": [Function],
          "responseOverride": undefined,
          "shouldRefetch": [Function],
          "startFetchAction": [Function],
          "url": "https://cms.integreat-app.de/{location}/de/wp-json/extensions/v0/modified_content/events?since=1970-01-01T00:00:00Z",
        },
      ]
    }
  >
    <Connect(I18nProvider)>
      <RouterFragment
        routerConfig={
          RouteConfig {
            "routes": Array [
              Route {
                "id": null,
                "path": "/",
                "pattern": UrlPattern {
                  "ast": Array [
                     {
                      "tag": "static",
                      "value": "/",
                    },
                  ],
                  "isRegex": false,
                  "names": Array [],
                  "regex": /\\^\\\\/\\$/,
                },
              },
              Route {
                "id": [Function],
                "path": "/:location/:language/search",
                "pattern": UrlPattern {
                  "ast": Array [
                     {
                      "tag": "static",
                      "value": "/",
                    },
                     {
                      "tag": "named",
                      "value": "location",
                    },
                     {
                      "tag": "static",
                      "value": "/",
                    },
                     {
                      "tag": "named",
                      "value": "language",
                    },
                     {
                      "tag": "static",
                      "value": "/search",
                    },
                  ],
                  "isRegex": false,
                  "names": Array [
                    "location",
                    "language",
                  ],
                  "regex": /\\^\\\\/\\(\\[a-zA-Z0-9-_~ %\\]\\+\\)\\\\/\\(\\[a-zA-Z0-9-_~ %\\]\\+\\)\\\\/search\\$/,
                },
              },
              Route {
                "id": [Function],
                "path": "/:location/:language/disclaimer",
                "pattern": UrlPattern {
                  "ast": Array [
                     {
                      "tag": "static",
                      "value": "/",
                    },
                     {
                      "tag": "named",
                      "value": "location",
                    },
                     {
                      "tag": "static",
                      "value": "/",
                    },
                     {
                      "tag": "named",
                      "value": "language",
                    },
                     {
                      "tag": "static",
                      "value": "/disclaimer",
                    },
                  ],
                  "isRegex": false,
                  "names": Array [
                    "location",
                    "language",
                  ],
                  "regex": /\\^\\\\/\\(\\[a-zA-Z0-9-_~ %\\]\\+\\)\\\\/\\(\\[a-zA-Z0-9-_~ %\\]\\+\\)\\\\/disclaimer\\$/,
                },
              },
              Route {
                "id": [Function],
                "path": "/:location/:language/events(/:id)",
                "pattern": UrlPattern {
                  "ast": Array [
                     {
                      "tag": "static",
                      "value": "/",
                    },
                     {
                      "tag": "named",
                      "value": "location",
                    },
                     {
                      "tag": "static",
                      "value": "/",
                    },
                     {
                      "tag": "named",
                      "value": "language",
                    },
                     {
                      "tag": "static",
                      "value": "/events",
                    },
                     {
                      "tag": "optional",
                      "value": Array [
                         {
                          "tag": "static",
                          "value": "/",
                        },
                         {
                          "tag": "named",
                          "value": "id",
                        },
                      ],
                    },
                  ],
                  "isRegex": false,
                  "names": Array [
                    "location",
                    "language",
                    "id",
                  ],
                  "regex": /\\^\\\\/\\(\\[a-zA-Z0-9-_~ %\\]\\+\\)\\\\/\\(\\[a-zA-Z0-9-_~ %\\]\\+\\)\\\\/events\\(\\?:\\\\/\\(\\[a-zA-Z0-9-_~ %\\]\\+\\)\\)\\?\\$/,
                },
              },
              Route {
                "id": [Function],
                "path": "/:location/:language/fetch-pdf",
                "pattern": UrlPattern {
                  "ast": Array [
                     {
                      "tag": "static",
                      "value": "/",
                    },
                     {
                      "tag": "named",
                      "value": "location",
                    },
                     {
                      "tag": "static",
                      "value": "/",
                    },
                     {
                      "tag": "named",
                      "value": "language",
                    },
                     {
                      "tag": "static",
                      "value": "/fetch-pdf",
                    },
                  ],
                  "isRegex": false,
                  "names": Array [
                    "location",
                    "language",
                  ],
                  "regex": /\\^\\\\/\\(\\[a-zA-Z0-9-_~ %\\]\\+\\)\\\\/\\(\\[a-zA-Z0-9-_~ %\\]\\+\\)\\\\/fetch\\\\-pdf\\$/,
                },
              },
              Route {
                "id": [Function],
                "path": "/:location/:language(/*)",
                "pattern": UrlPattern {
                  "ast": Array [
                     {
                      "tag": "static",
                      "value": "/",
                    },
                     {
                      "tag": "named",
                      "value": "location",
                    },
                     {
                      "tag": "static",
                      "value": "/",
                    },
                     {
                      "tag": "named",
                      "value": "language",
                    },
                     {
                      "tag": "optional",
                      "value": Array [
                         {
                          "tag": "static",
                          "value": "/",
                        },
                         {
                          "tag": "wildcard",
                          "value": "*",
                        },
                      ],
                    },
                  ],
                  "isRegex": false,
                  "names": Array [
                    "location",
                    "language",
                    "_",
                  ],
                  "regex": /\\^\\\\/\\(\\[a-zA-Z0-9-_~ %\\]\\+\\)\\\\/\\(\\[a-zA-Z0-9-_~ %\\]\\+\\)\\(\\?:\\\\/\\(\\.\\*\\?\\)\\)\\?\\$/,
                },
              },
              Route {
                "id": [Function],
                "path": "/disclaimer",
                "pattern": UrlPattern {
                  "ast": Array [
                     {
                      "tag": "static",
                      "value": "/disclaimer",
                    },
                  ],
                  "isRegex": false,
                  "names": Array [],
                  "regex": /\\^\\\\/disclaimer\\$/,
                },
              },
              Route {
                "id": null,
                "path": "/:unknown(/)",
                "pattern": UrlPattern {
                  "ast": Array [
                     {
                      "tag": "static",
                      "value": "/",
                    },
                     {
                      "tag": "named",
                      "value": "unknown",
                    },
                     {
                      "tag": "optional",
                      "value": Array [
                         {
                          "tag": "static",
                          "value": "/",
                        },
                      ],
                    },
                  ],
                  "isRegex": false,
                  "names": Array [
                    "unknown",
                  ],
                  "regex": /\\^\\\\/\\(\\[a-zA-Z0-9-_~ %\\]\\+\\)\\(\\?:\\\\/\\)\\?\\$/,
                },
              },
              Route {
                "id": [Function],
                "path": "/:language(/)",
                "pattern": UrlPattern {
                  "ast": Array [
                     {
                      "tag": "static",
                      "value": "/",
                    },
                     {
                      "tag": "named",
                      "value": "language",
                    },
                     {
                      "tag": "optional",
                      "value": Array [
                         {
                          "tag": "static",
                          "value": "/",
                        },
                      ],
                    },
                  ],
                  "isRegex": false,
                  "names": Array [
                    "language",
                  ],
                  "regex": /\\^\\\\/\\(\\[a-zA-Z0-9-_~ %\\]\\+\\)\\(\\?:\\\\/\\)\\?\\$/,
                },
              },
            ],
          }
        }
      />
    </Connect(I18nProvider)>
  </EndpointProvider>
</Provider>
`;
